Assets/_BForBoss/_Core/Scripts/Boss/DerekHealthBehaviour.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using Perigon.Weapons;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public class DerekHealthBehaviour : MonoBehaviour, IBulletCollision
+    {
+        [SerializeField] private int _maxHealth = 12;
+        [SerializeField] private int _damagePerShot = 1;
+
+        private int _elapsedHealth;
+        private float _threshold;
+        private DerekContextManager.Vulnerability _vulnerabilityState = DerekContextManager.Vulnerability.Invulnerable;
+        
+        private event Action<DerekHealthViewState> OnStateChanged;
+        private event Action OnThresholdHit;
+
+        public DerekContextManager.Vulnerability VulnerabilityState
+        {
+            get => _vulnerabilityState;
+            set
+            {
+                _vulnerabilityState = value;
+                OnStateChanged?.Invoke(MapToState());
+            } 
+        }
+
+        public void Initialize(
```

Comment:
So ideally here. You would connect everything together in the context manager.

DerekHealthBehaviour  - [SerializeField] within the context manager
and then Initialize via
DerekHealthBehaviour.Initialize(onThresholdHit: { }, onStateChanged: { view.Set(state)})

Reply:
I think this should be connected via the BossManager, not the ContextManager. But yeah, the functionality sounds good!